test/integration/api/v1/registrant/contacts/list_test.rb

```diff
@@ -55,6 +56,42 @@ def test_returns_direct_contacts_when_company_register_is_unavailable
     assert_equal '1234', response_json.first[:ident][:code]
   end
 
+  def test_out_of_range_limit
+    get api_v1_registrant_contacts_path + "?limit=300", as: :json, headers: { 'HTTP_AUTHORIZATION' => auth_token }
+    response_json = JSON.parse(response.body, symbolize_names: true)
+
+    text_response = JSON.pretty_generate(response_json[:errors][0][:limit][0])
+
+    assert_equal text_response, '"parameter is out of range"'
+  end
+
+  def test_negative_offset
+    get api_v1_registrant_contacts_path + "?offset=-300", as: :json, headers: { 'HTTP_AUTHORIZATION' => auth_token }
+    response_json = JSON.parse(response.body, symbolize_names: true)
+
+    text_response = JSON.pretty_generate(response_json[:errors][0][:offset][0])
+
+    assert_equal text_response, '"parameter is out of range"'
+  end
+
+  def test_show_valid_contact
+    get api_v1_registrant_contacts_path + "/eb2f2766-b44c-4e14-9f16-32ab1a7cb957", as: :json, headers: { 'HTTP_AUTHORIZATION' => auth_token }
+    response_json = JSON.parse(response.body, symbolize_names: true)
+
+    text_response = response_json[:name]
+
+    assert_equal @contact[:name], text_response
+  end
+
+  def test_show_invalid_contact
+    get api_v1_registrant_contacts_path + "/435", as: :json, headers: { 'HTTP_AUTHORIZATION' => auth_token }
+    response_json = JSON.parse(response.body, symbolize_names: true)
+
+    text_response = response_json[:errors][0][:base][0]
+
+    assert_equal text_response, 'Contact not found'
+  end
+
   private
```

Comment:
BTW have a look on it - that's how we are separating helpers from test methods - by setting them as private.